galsim/config/output.py

```
@@ -467,22 +500,47 @@ def getNImages(self, config, base, file_num):
         """
         return 1
 
-    def writeFile(self, data, file_name):
+    def canAddHdus(self):
+        """Returns whether it is permissible to add extra HDUs to the end of the data list.
+
+        In the base class, this returns True.
+        """
+        return True
+
+    def addExtraOutputHDUs(self, config, data, logger):
+        """If appropriate, add any extra output items that go into HDUs to the data list.
+
+        @param config           The configuration dict for the output field.
+        @param data             The data to write.  Usually a list of images.
+        @param logger           If given, a logger object to log progress.
+
+        @returns data (possibly updated with additional items)
+        """
+        if self.canAddHdus():
+            data = galsim.config.AddExtraOutputHDUs(config, data, logger)
```

Yes.  In fact, I think it should raise an exception.  Added.  5bd542acf